src/placeholder/parameter/prior.py

```diff
@@ -0,0 +1,69 @@
+"""
+===================
+Prior Distributions
+===================
+
+Gives prior specifications that can be attached to fixed
+and/or random effect components of a variable and may be used
+in the solver optimization.
+
+The default prior only includes upper and lower bounds for box constraints and defaults
+to :python:`[-np.inf, np.inf]`. Alternative priors include
+:class:`~placeholder.parameter.prior.GaussianPrior`.
+"""
+
+import numpy as np
+from dataclasses import dataclass
+
+from placeholder.exceptions import PlaceholderError
+
+
+class PriorError(PlaceholderError):
+    pass
+
+
+def _check_consistency(x, y):
+    if isinstance(x, List) or isinstance(y, List):
+        if not (isinstance(x, List) or isinstance(y, List)):
+            raise PriorError(f"{x.__name__} and {y.__name__} are not of the same type.")
+        if not len(x) == len(y):
+            raise PriorError(f"{x.__name__} and {y.__name__} are not of the same length.")
+
+
+@dataclass
+class Prior:
+
```

Comment:
just wonder if there is some relationship between this `Prior` class and the `Likelihood` class we were talking about. They seem to share some similarities to me (or maybe not at all! but I was just wondering.)

Reply:
I think it's a different class but yes to a relationship!